src/app/core/signing.service.ts

```diff
@@ -0,0 +1,85 @@
+import { Injectable } from '@angular/core';
+import { Fonts } from '../shared/interfaces';
+import { ActivatedRoute, Router } from '@angular/router';
+import { AuthService } from './auth.service';
+import { MsgIoService } from './msg-io.service';
+import { FontService } from './font.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class SigningService {
+  public signingBoxText;
+  public backgroundColor;
+  public textColor;
+  public canSend;
+  public charCount;
+  public maxCharCount: number;
+  public charCountColor;
+
+  // font service stuff
+  public currentFont; // Used to select a font from fontDict
+  public currentFontFamily; // Used to CSS select the font
+  public currentFontDisplay; // Shown in the Font Dropdown menu
+  public fontDict: Fonts;
+
+  constructor() {
+    this.fontDict = FontService.fonts;
+    this.maxCharCount = MsgIoService.maxMessageLength;
+  }
+
+  /**
+   * Sets all class variables to their default values
+   */
+  resetDefaultValues() {
```

Comment:
Why is this a separate function?

Reply:
We want any component to be able to reset the values of all the message properties.
If someone plays with the values on Home, we don't want that to show up on Signing when they go to sign, since the service is a singleton